exporters/prometheus/go.mod

```diff
@@ -3,7 +3,7 @@ module go.opentelemetry.io/otel/exporters/prometheus
 go 1.23.0
 
 require (
-	github.com/prometheus/client_golang v1.20.5
+	github.com/prometheus/client_golang v1.21.1
```

Comment:
This upgrade has breaking changes (see https://github.com/open-telemetry/opentelemetry-go/pull/6349)
So could we avoid having it within change? It should either be its own PR, or not happen.

Reply:
Hey! Thanks for the review. No, this is not possible as the functionality depends on a new function that is only available in this version.

Could you please elaborate on what is the breaking change in client_golang? https://github.com/prometheus/client_golang/pull/1712 is it this one?

Reply:
Attributes are changing, switching from underscores to dots.
I'm not saying we shouldn't upgrade. But we should separate the upgrade from this new feature.

And ideally, we should provide a (prometheus native) way to keep the old behavior for attributes names.

Reply:
I had run into this change as well, I realised that the behaviour changed in assertions only, i.e. `_` is replaced with `.`, but this change doesn't reflect in http output via the [`promhttp.HandlerFor`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp#HandlerFor).

However, this should be verified, and I think it makes sense to upgrade `prometheus/client_golang` in a separate PR.

Reply:
Opened the PR https://github.com/open-telemetry/opentelemetry-go/pull/6433